<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SolidRocketEngineWidget</class>
 <widget class="QWidget" name="SolidRocketEngineWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>515</width>
    <height>616</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string/>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="2">
       <widget class="QLineEdit" name="massLineEdit"/>
      </item>
      <item row="5" column="2">
       <widget class="QLineEdit" name="rampUpTimeLineEdit">
        <property name="toolTip">
         <string>The time it takes for the engine to go from zero thrust to its initial rated thrust</string>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QLabel" name="rampUpTimeUnitLabel">
        <property name="text">
         <string>s</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="rampUpLabel">
        <property name="text">
         <string>Ramp-up Time</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="rampDownLabel">
        <property name="text">
         <string>Ramp-down Time</string>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="massUnitLabel">
          <property name="text">
           <string>lbs</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="massComboBox"/>
        </item>
       </layout>
      </item>
      <item row="6" column="3">
       <widget class="QLabel" name="rampDownTimeUnitLabel">
        <property name="text">
         <string>s</string>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <widget class="QLineEdit" name="rampDownTimeLineEdit">
        <property name="toolTip">
         <string>The time it takes for the engine to go from its final rated thrust to zero thrust</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="massLabel">
          <property name="text">
           <string>Mass</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>85</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Specific Impulse</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="5">
       <widget class="QLineEdit" name="specificImpulseSeaLevelLineEdit">
        <property name="toolTip">
         <string>The specific impulse at sea level</string>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="specificImpulseSeaLevelUnitLabel">
          <property name="text">
           <string>s</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>75</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="specificImpulseSeaLevelLabel">
          <property name="text">
           <string>Sea Level</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>62</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="1" column="5">
       <widget class="QLineEdit" name="specificImpulseVacuumLineEdit">
        <property name="toolTip">
         <string>The specific impulse at vacuum conditions</string>
        </property>
       </widget>
      </item>
      <item row="1" column="6">
       <widget class="QLabel" name="specificImpulseVacuumUnitLabel">
        <property name="text">
         <string>s</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Vacuum</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="specificationsGroupBox">
     <property name="title">
      <string/>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QComboBox" name="altitudeComboBox">
        <property name="toolTip">
         <string>Specifies which altitude condition to use when calculating thrust, burn time, or propellant mass</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="specificationsComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>The two parameters to specify. The remaining parameter is calculated from the other two.</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QLabel" name="maxRampDownThrustUnitLabel">
        <property name="text">
         <string>lbs</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="altitudeLabel">
        <property name="text">
         <string>Altitude Condition</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="maxRampDownThrustLineEdit">
        <property name="toolTip">
         <string>The rated thrust just before ramp-down</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="propellantMassLabel">
        <property name="text">
         <string>Propellant Mass</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="maxRampDownThrustLabel">
        <property name="text">
         <string>Rated Thrust, Final</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="specificationsLabel">
        <property name="text">
         <string>Specified Characteristics</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="propellantMassLineEdit">
        <property name="toolTip">
         <string>The amount of propellant. Propellant fraction is given by the propellant mass divided by the empty mass</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="maxRampUpThrustLineEdit">
        <property name="toolTip">
         <string>The rated thrust after the initial ramp-up</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="maxRampUpThrustLabel">
        <property name="text">
         <string>Rated Thrust, Initial</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="burnTimeLineEdit"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="burnTimeLabel">
        <property name="text">
         <string>Burn Time</string>
        </property>
       </widget>
      </item>
      <item row="6" column="2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="propellantMassUnitLabel">
          <property name="text">
           <string>lbs</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="propellantMassComboBox"/>
        </item>
       </layout>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="maxRampUpThrustUnitLabel">
        <property name="text">
         <string>lbs</string>
        </property>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QLabel" name="burnTimeUnitLabel">
        <property name="text">
         <string>s</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>massLineEdit</tabstop>
  <tabstop>massComboBox</tabstop>
  <tabstop>rampUpTimeLineEdit</tabstop>
  <tabstop>rampDownTimeLineEdit</tabstop>
  <tabstop>altitudeComboBox</tabstop>
  <tabstop>specificationsComboBox</tabstop>
  <tabstop>maxRampUpThrustLineEdit</tabstop>
  <tabstop>maxRampDownThrustLineEdit</tabstop>
  <tabstop>burnTimeLineEdit</tabstop>
  <tabstop>propellantMassLineEdit</tabstop>
  <tabstop>propellantMassComboBox</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
